### PR TITLE
[13.0][FIX] sale, rename hide_expense_policy field

### PR DESCRIPTION
**Pull request by GSLabIt**
Description of the issue/feature this PR addresses: https://github.com/OCA/OpenUpgrade/issues/3456

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Can you modify the analysis work file?

**Comment by GSLabIt** [comment]
> Can you modify the analysis work file?

Manually? or is there a script to run? And what to add exactly?

**Comment by pedrobaeza** [comment]
Manually. There we take note of the changes and the actions done. There should be one line saying that the field is added and removed. Join both and say that you are renaming it.

**Comment by pedrobaeza** [comment]
Please always squash commits that are related to the same thing, and the commit message can be:  `[OU-FIX] sale: rename hide_expense_policy field`

File: addons/sale/migrations/13.0.1.1/openupgrade_analysis_work.txt [hunk 1]
```diff
@@ -29,6 +29,13 @@ sale         / crm.team                 / use_invoices (boolean)        : DEL
 
 sale         / product.attribute        / display_type (selection)      : NEW required, selection_keys: ['color', 'radio', 'select'], req_default: function, hasdefault
 sale         / product.attribute        / type (selection)              : DEL required, selection_keys: ['color', 'radio', 'select'], req_default: function
+
+sale         / product.template        / visible_expense_policy (Boolean) : NEW
```
Comment (pedrobaeza) [inline addons/sale/migrations/13.0.1.1/openupgrade_analysis_work.txt:33]: Aren't they in the analysis file? Are you sure this is something of this version, or is it on a prior one?

Comment (GSLabIt) [reply addons/sale/migrations/13.0.1.1/openupgrade_analysis_work.txt:33]: No, they aren't in analysis files, for some reason out of my control they are missed.  

Actually what i'm doing is a migration from `12.0` to `13.0` using branch `13.0`, am i doing right? 

If so, from odoo `12.0` to `13.0`, they changed the name of `hide_expense_policy` to `visible_expense_policy`. Running openupgrade instance it raise the below error. Renaming the fields will make it run fine.

```
2022-07-15 08:51:48,722 8452 INFO sicur13 odoo.modules.migration: module sale: Running migration [>13.0.1.1] pre-migration 
2022-07-15 08:51:48,838 8452 INFO sicur13 OpenUpgrade: sale: pre-migration script called with version 12.0.1.1 
2022-07-15 08:51:48,839 8452 INFO sicur13 OpenUpgrade: table sale_order_line_invoice_rel: renaming to ou_sale_order_line_invoice_rel 
2022-07-15 08:51:48,844 8452 INFO sicur13 OpenUpgrade: table product_attribute, column type: renaming to display_type 
2022-07-15 08:51:48,865 8452 INFO sicur13 OpenUpgrade: table product_attribute_custom_value, column attribute_value_id: renaming to openupgrade_legacy_13_0_attribute_value_id 
2022-07-15 08:51:48,881 8452 DEBUG sicur13 OpenUpgrade: 614 rows affected after 0:00:00.014070 running 
                ALTER TABLE sale_order
                ADD COLUMN openupgrade_legacy_13_0_date_order timestamp without time zone;
                UPDATE sale_order SET openupgrade_legacy_13_0_date_order=date_order;
                 
2022-07-15 08:51:48,881 8452 DEBUG sicur13 OpenUpgrade: 1 rows affected after 0:00:00.000121 running UPDATE ir_model_data SET module = 'sale', name = 'access_account_move_manager' WHERE module = 'sale' and name = 'access_account_invoice_manager' 
2022-07-15 08:51:48,882 8452 DEBUG sicur13 OpenUpgrade: 1 rows affected after 0:00:00.000122 running UPDATE ir_model_data SET module = 'sale', name = 'access_account_move_salesman' WHERE module = 'sale' and name = 'access_account_invoice_salesman' 
2022-07-15 08:51:48,882 8452 INFO sicur13 OpenUpgrade: Moving fields campaign_id, medium_id, source_id in model sale.order from module 'sale_crm' to module 'sale' 
2022-07-15 08:51:48,882 8452 DEBUG sicur13 OpenUpgrade: 0 rows affected after 0:00:00.000687 running 
        UPDATE ir_model_data imd
        SET module = 'sale'
        FROM ir_model_fields imf
        WHERE
            imf.model = 'sale.order' AND
            imf.name IN ('campaign_id', 'medium_id', 'source_id') AND
            imd.module = 'sale_crm' AND
            imd.model = 'ir.model.fields' AND
            imd.res_id = imf.id AND
            imd.id NOT IN (
               SELECT id FROM ir_model_data WHERE module = 'sale'
            )
         
2022-07-15 08:51:48,883 8452 DEBUG sicur13 OpenUpgrade: 0 rows affected after 0:00:00.000865 running 
        UPDATE ir_translation it
        SET module = 'sale'
        FROM ir_model_fields imf
        WHERE
            imf.model = 'sale.order' AND
            imf.name IN ('campaign_id', 'medium_id', 'source_id') AND
            it.res_id = imf.id AND
            it.module = 'sale_crm' AND ((
                it.name LIKE 'ir.model.fields,field_%' AND
                it.type = 'model'
            ) OR (
                it.type = 'field'
            ))
         
2022-07-15 08:51:48,884 8452 DEBUG sicur13 OpenUpgrade: -1 rows affected after 0:00:00.000154 running ALTER TABLE "utm_campaign" ADD COLUMN "company_id" int4 
2022-07-15 08:51:48,885 8452 DEBUG sicur13 OpenUpgrade: 1 rows affected after 0:00:00.000774 running 
                    INSERT INTO ir_model_fields (
                        model_id, model, name, field_description,
                        ttype, state
                    ) VALUES (
                        317, 'utm.campaign', 'company_id',
                        'OU', 'many2one',
                        'base'
                    ) RETURNING id
                     
2022-07-15 08:51:48,885 8452 DEBUG sicur13 OpenUpgrade: 1 rows affected after 0:00:00.000257 running UPDATE crm_team SET invoiced_target = 0 WHERE NOT use_invoices 
2022-07-15 08:51:49,287 8452 INFO sicur13 odoo.modules.registry: module sale: creating or updating database tables 
2022-07-15 08:51:50,563 8452 INFO sicur13 odoo.modules.loading: loading sale/security/sale_security.xml 
2022-07-15 08:51:51,347 8452 INFO sicur13 odoo.modules.loading: loading sale/security/ir.model.access.csv 
2022-07-15 08:51:51,564 8452 INFO sicur13 odoo.modules.loading: loading sale/report/sale_report.xml 
2022-07-15 08:51:51,578 8452 INFO sicur13 odoo.modules.loading: loading sale/report/sale_report_views.xml 
2022-07-15 08:51:51,647 8452 INFO sicur13 odoo.modules.loading: loading sale/report/sale_report_templates.xml 
2022-07-15 08:51:51,727 8452 INFO sicur13 odoo.modules.loading: loading sale/report/invoice_report_templates.xml 
2022-07-15 08:51:51,777 8452 INFO sicur13 odoo.modules.loading: loading sale/report/report_all_channels_sales_views.xml 
2022-07-15 08:51:51,785 8452 INFO sicur13 odoo.modules.loading: loading sale/data/ir_sequence_data.xml 
2022-07-15 08:51:51,790 8452 INFO sicur13 odoo.modules.loading: loading sale/data/mail_data.xml 
2022-07-15 08:51:51,814 8452 INFO sicur13 odoo.modules.loading: loading sale/data/sale_data.xml 
2022-07-15 08:51:51,825 8452 INFO sicur13 odoo.modules.loading: loading sale/wizard/sale_make_invoice_advance_views.xml 
2022-07-15 08:51:51,886 8452 INFO sicur13 odoo.modules.loading: loading sale/views/sale_views.xml 
2022-07-15 08:51:52,400 8452 INFO sicur13 odoo.addons.base.models.ir_ui_view: Il campo `hide_expense_policy` non esiste

Contesto errore:
Vista `product.template.product.website.form`
[view_id: 4312, xml_id: sale.product_template_form_view, model: product.template, parent_id: 340] 
2022-07-15 08:51:52,415 8452 ERROR sicur13 odoo.modules.registry: Failed to load registry 
Traceback (most recent call last):
  File "/openupgrade13/OpenUpgrade/odoo/tools/convert.py", line 717, in parse
    self._tag_root(de)
  File "/openupgrade13/OpenUpgrade/odoo/tools/convert.py", line 679, in _tag_root
    f(rec)
  File "/openupgrade13/OpenUpgrade/odoo/tools/convert.py", line 582, in _tag_record
    record = model._load_records([data], self.mode == 'update')
  File "/openupgrade13/OpenUpgrade/odoo/models.py", line 4171, in _load_records
    data['record']._load_records_write(data['values'])
  File "/openupgrade13/OpenUpgrade/odoo/addons/base/models/ir_ui_view.py", line 1409, in _load_records_write
    super(View, self)._load_records_write(values)
  File "/openupgrade13/OpenUpgrade/odoo/models.py", line 4101, in _load_records_write
    self.write(values)
  File "/openupgrade13/OpenUpgrade/addons/website_theme_install/models/theme_models.py", line 226, in write
    res = super(IrUiView, other_views).write(vals)
  File "/openupgrade13/OpenUpgrade/addons/website/models/ir_ui_view.py", line 51, in write
    return super(View, self).write(vals)
  File "/openupgrade13/OpenUpgrade/odoo/addons/base/models/ir_ui_view.py", line 494, in write
    res = super(View, self).write(self._compute_defaults(vals))
  File "/openupgrade13/OpenUpgrade/odoo/models.py", line 3655, in write
    fields[0].determine_inverse(real_recs)
  File "/openupgrade13/OpenUpgrade/odoo/fields.py", line 1156, in determine_inverse
    getattr(records, self.inverse)()
  File "/openupgrade13/OpenUpgrade/odoo/addons/base/models/ir_ui_view.py", line 299, in _inverse_arch
    view.write(data)
  File "/openupgrade13/OpenUpgrade/addons/website_theme_install/models/theme_models.py", line 226, in write
    res = super(IrUiView, other_views).write(vals)
  File "/openupgrade13/OpenUpgrade/addons/website/models/ir_ui_view.py", line 51, in write
    return super(View, self).write(vals)
  File "/openupgrade13/OpenUpgrade/odoo/addons/base/models/ir_ui_view.py", line 494, in write
    res = super(View, self).write(self._compute_defaults(vals))
  File "/openupgrade13/OpenUpgrade/odoo/models.py", line 3650, in write
    real_recs._validate_fields(vals, inverse_fields)
  File "/openupgrade13/OpenUpgrade/odoo/models.py", line 1216, in _validate_fields
    check(self)
  File "/openupgrade13/OpenUpgrade/odoo/addons/base/models/ir_ui_view.py", line 394, in _check_xml
    self.postprocess_and_fields(view.model, view_doc, view.id)
  File "/openupgrade13/OpenUpgrade/odoo/addons/base/models/ir_ui_view.py", line 1007, in postprocess_and_fields
    msg_lines.append(msg_fmt % name)
TypeError: %u format: a number is required, not str

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/openupgrade13/OpenUpgrade/odoo/modules/registry.py", line 87, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/openupgrade13/OpenUpgrade/odoo/modules/loading.py", line 489, in load_modules
    force, status, report, loaded_modules, update_module, models_to_check, upg_registry)
  File "/openupgrade13/OpenUpgrade/odoo/modules/loading.py", line 368, in load_marked_modules
    upg_registry=upg_registry,
  File "/openupgrade13/OpenUpgrade/odoo/modules/loading.py", line 248, in load_module_graph
    load_data(cr, idref, mode, kind='data', package=package, report=report)
  File "/openupgrade13/OpenUpgrade/odoo/modules/loading.py", line 72, in load_data
    tools.convert_file(cr, package.name, filename, idref, mode, noupdate, kind, report)
  File "/openupgrade13/OpenUpgrade/odoo/tools/convert.py", line 741, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/openupgrade13/OpenUpgrade/odoo/tools/convert.py", line 808, in convert_xml_import
    obj.parse(doc.getroot())
  File "/openupgrade13/OpenUpgrade/odoo/tools/convert.py", line 726, in parse
    exc_info[2]
  File "/openupgrade13/OpenUpgrade/odoo/tools/pycompat.py", line 13, in reraise
    raise value.with_traceback(tb)
  File "/openupgrade13/OpenUpgrade/odoo/tools/convert.py", line 717, in parse
    self._tag_root(de)
  File "/openupgrade13/OpenUpgrade/odoo/tools/convert.py", line 679, in _tag_root
    f(rec)
  File "/openupgrade13/OpenUpgrade/odoo/tools/convert.py", line 582, in _tag_record
    record = model._load_records([data], self.mode == 'update')
  File "/openupgrade13/OpenUpgrade/odoo/models.py", line 4171, in _load_records
    data['record']._load_records_write(data['values'])
  File "/openupgrade13/OpenUpgrade/odoo/addons/base/models/ir_ui_view.py", line 1409, in _load_records_write
    super(View, self)._load_records_write(values)
  File "/openupgrade13/OpenUpgrade/odoo/models.py", line 4101, in _load_records_write
    self.write(values)
  File "/openupgrade13/OpenUpgrade/addons/website_theme_install/models/theme_models.py", line 226, in write
    res = super(IrUiView, other_views).write(vals)
  File "/openupgrade13/OpenUpgrade/addons/website/models/ir_ui_view.py", line 51, in write
    return super(View, self).write(vals)
  File "/openupgrade13/OpenUpgrade/odoo/addons/base/models/ir_ui_view.py", line 494, in write
    res = super(View, self).write(self._compute_defaults(vals))
  File "/openupgrade13/OpenUpgrade/odoo/models.py", line 3655, in write
    fields[0].determine_inverse(real_recs)
  File "/openupgrade13/OpenUpgrade/odoo/fields.py", line 1156, in determine_inverse
    getattr(records, self.inverse)()
  File "/openupgrade13/OpenUpgrade/odoo/addons/base/models/ir_ui_view.py", line 299, in _inverse_arch
    view.write(data)
  File "/openupgrade13/OpenUpgrade/addons/website_theme_install/models/theme_models.py", line 226, in write
    res = super(IrUiView, other_views).write(vals)
  File "/openupgrade13/OpenUpgrade/addons/website/models/ir_ui_view.py", line 51, in write
    return super(View, self).write(vals)
  File "/openupgrade13/OpenUpgrade/odoo/addons/base/models/ir_ui_view.py", line 494, in write
    res = super(View, self).write(self._compute_defaults(vals))
  File "/openupgrade13/OpenUpgrade/odoo/models.py", line 3650, in write
    real_recs._validate_fields(vals, inverse_fields)
  File "/openupgrade13/OpenUpgrade/odoo/models.py", line 1216, in _validate_fields
    check(self)
  File "/openupgrade13/OpenUpgrade/odoo/addons/base/models/ir_ui_view.py", line 394, in _check_xml
    self.postprocess_and_fields(view.model, view_doc, view.id)
  File "/openupgrade13/OpenUpgrade/odoo/addons/base/models/ir_ui_view.py", line 1007, in postprocess_and_fields
    msg_lines.append(msg_fmt % name)
odoo.tools.convert.ParseError: "%u format: a number is required, not str" while parsing /openupgrade13/OpenUpgrade/addons/sale/views/sale_views.xml:2, near
```

Comment (pedrobaeza) [reply addons/sale/migrations/13.0.1.1/openupgrade_analysis_work.txt:33]: It's weird that they are not reflected, but I see that the data model is indeed changed that way.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@MiquelRForgeFlow do you have any explanation why the fields are not detected?

**Comment by MiquelRForgeFlow** [comment]
Both hide_expense_policy (v12) and visible_expense_policy (v13) are non-stored fields. That's why they don't appear in the analysis.

**Comment by pedrobaeza** [comment]
> Both hide_expense_policy (v12) and visible_expense_policy (v13) are non-stored fields. That's why they don't appear in the analysis.

I never remember this...

For the `rename_fields` to avoid errors in filters/exports may be interesting to have them (although I remember this to be discussed before...).

**Comment by MiquelRForgeFlow** [comment]
It's better to ignore them. If you find a case like this, then do the fix like this one and that's all. They are supercorner cases.

**Comment by GSLabIt** [comment]
@pedrobaeza @MiquelRForgeFlow i'm still experiencing the issue. It seems that field is properly renamed on model but not in the view, actually in `sale.product_template_form_view_invoice_policy`:

https://github.com/odoo/odoo/blob/67461b4ca1a9c2eb8ea1c0cc4bc39d91cf0d2823/addons/sale/views/sale_views.xml#L1022-L1035

Commenting out the 2 lines referencing `hide_expense_policy` will do the job. 

Any hint on how to fix the migration script to have field name changed?

Thanks

**Comment by pedrobaeza** [comment]
Remove the offending view before starting the migration process.

**Comment by GSLabIt** [comment]
> Remove the offending view before starting the migration process.

K, thanks.

**Pull request merged**